cmd/cmd.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cmd
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/ubuntu-core/snappy/osutil"
+	"github.com/ubuntu-core/snappy/release"
+)
+
+const key = "SNAP_INNER"
+
+// SwitchToTheRealOne makes sure you're executing the "real"
+// binary. I.e. the one that ships in the ubuntu-core snap.
+func SwitchToTheRealOne() {
+	if !release.OnClassic {
+		// you're already the real deal, natch
+		return
+	}
+
+	if os.Getenv(key) != "" {
```

Comment:
Can we please check for "1" specifically?  It's generally a bug to consider the existence of a variable at all to be indicative of a given choice, as it's an obvious thing to invert the logic (set it to `0`) and expect it to invert the semantic too.